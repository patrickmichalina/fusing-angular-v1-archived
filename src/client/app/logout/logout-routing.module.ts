import { LogoutComponent } from './logout.component'
import { NgModule } from '@angular/core'
import { RouterModule } from '@angular/router'

@NgModule({
  imports: [
    RouterModule.forChild([
      {
        path: '',
        component: LogoutComponent,
        data: {
          meta: {
            title: 'Logout'
          }
        }
      }
    ])
  ],
  exports: [RouterModule]
})
export class LogoutRoutingModule {}
