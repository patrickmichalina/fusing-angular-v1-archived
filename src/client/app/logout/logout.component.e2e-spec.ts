import { baseUrl, browser } from '../../../../tools/test/jest.e2e-setup'

describe('Logout Page', () => {
  it('should have title', async () => {
    const page = browser.goto(`${baseUrl}/logout`)
    const text = await page.evaluate(() => document.title)
    expect(text).toContain('Home')
  })
})
