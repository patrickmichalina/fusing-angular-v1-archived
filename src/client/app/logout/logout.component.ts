import { ChangeDetectionStrategy, Component } from '@angular/core'
import { AuthService } from '../shared/services/auth.service'
import { Router } from '@angular/router'

@Component({
  selector: 'pm-logout',
  templateUrl: './logout.component.html',
  styleUrls: ['./logout.component.css'],
  changeDetection: ChangeDetectionStrategy.OnPush
})
export class LogoutComponent {
  constructor(auth: AuthService, router: Router) {
    auth.logout()
    router.navigate(['/'], { skipLocationChange: true })
  }
}
