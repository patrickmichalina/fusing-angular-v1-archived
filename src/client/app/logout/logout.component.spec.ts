import { LogoutComponent } from './logout.component'
import { TransferState } from '@angular/platform-browser'
import { async, ComponentFixture, TestBed } from '@angular/core/testing'
import { Component } from '@angular/core'
import { LogoutModule } from './logout.module'
import { AppTestingModule } from '../../../testing/app-testing.module'
import { ResponseService } from '../shared/services/response.service'

@Component({
  selector: 'test-component',
  template: '<pm-logout></pm-logout>'
})
class TestComponent {}

describe(LogoutComponent.name, () => {
  let fixture: ComponentFixture<TestComponent>

  beforeEach(
    async(() => {
      TestBed.configureTestingModule({
        imports: [LogoutModule, AppTestingModule.forRoot()],
        declarations: [TestComponent],
        providers: [ResponseService, TransferState]
      }).compileComponents()
    })
  )

  beforeEach(
    async(() => {
      fixture = TestBed.createComponent(TestComponent)
    })
  )

  afterEach(
    async(() => {
      TestBed.resetTestingModule()
    })
  )

  it(
    'should compile',
    async(() => {
      fixture.detectChanges()
      expect(fixture.componentInstance).toBeDefined()
      expect(fixture).toMatchSnapshot()
    })
  )
})
