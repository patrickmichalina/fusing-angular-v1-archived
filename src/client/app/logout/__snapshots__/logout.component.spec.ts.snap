// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogoutComponent should compile 1`] = `
<test-component>
  <pm-logout>
    Logging out...
  </pm-logout>
</test-component>
`;
